src/tests/end-to-end/common/browser.ts

```diff
@@ -21,67 +23,70 @@ export class Browser {
         await this.underlyingBrowser.close();
     }
 
-    public async newPage(url: string): Promise<Page> {
-        const underlyingPage = await this.underlyingBrowser.newPage();
-        const page = new Page(underlyingPage, { onPageCrash: this.onPageCrash });
-        await page.goto(url);
-        this.pages.push(page);
-        return page;
-    }
-
-    public async setupNewTargetPage(): Promise<TargetPageController> {
-        const targetPage = await this.newTestResourcePage('all.html');
-
-        await targetPage.bringToFront();
-        const targetPageTabId = await this.getActivePageTabId();
-        return new TargetPageController(targetPage, targetPageTabId);
-    }
+    public async backgroundPage(): Promise<BackgroundPage> {
```

Comment:
I just though of this: while other "creational" functions here (like ` newPage` or ' newTargetPage'), `backgroundPage` is the only one that cached the result to return it on a subsequent call. I'm wondering if we should make that clear by renaming the function (maybe all the creational functions need to be renamed)

Reply:
This is the reason I named it `backgroundPage()` instead of `newBackgroundPage()` like the rest of them; I figured this was probably sufficient since I'd expect a typical maintainer of our code would probably be aware already that a background page is something that hangs around persistently, rather than being launched explictly. Do you have a specific verb in mind that might be more clear? `get` and `find` are the ones that came to my mind but I didn't think they were significantly more clear than `backgroundPage()` on its own...

Reply:
I thought about `get`, not sure if that's strong enough though. I guess the current difference may be enough.